pandas/indexes/base.py

```diff
@@ -165,7 +165,15 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
             if (issubclass(data.dtype.type, np.datetime64) or
                     is_datetimetz(data)):
                 from pandas.tseries.index import DatetimeIndex
-                result = DatetimeIndex(data, copy=copy, name=name, **kwargs)
+                if 'tz' in kwargs:
```

Comment:
we can use ``DatetimeIndex._simple_new(data, name=name, **kwargs)``. Needs to handle copy in advance.